airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/components/ReplicationView.tsx

```diff
@@ -1,32 +1,68 @@
 import { faSyncAlt } from "@fortawesome/free-solid-svg-icons";
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
-import { FormikHelpers } from "formik";
 import React, { useMemo, useState } from "react";
-import { FormattedMessage } from "react-intl";
-import { useAsyncFn } from "react-use";
+import { FormattedMessage, useIntl } from "react-intl";
+import { useAsyncFn, useUnmount } from "react-use";
 import styled from "styled-components";
 
-import { Button, Card } from "components";
+import { Button, Card, LabeledSwitch } from "components";
 import LoadingSchema from "components/LoadingSchema";
 
 import { toWebBackendConnectionUpdate } from "core/domain/connection";
-import { ConnectionStatus } from "core/request/AirbyteClient";
-import { useConfirmationModalService } from "hooks/services/ConfirmationModal";
+import { ConnectionStateType, ConnectionStatus } from "core/request/AirbyteClient";
+import { useModalService } from "hooks/services/Modal";
 import {
   useConnectionLoad,
-  useResetConnection,
+  useConnectionService,
   useUpdateConnection,
   ValuesProps,
 } from "hooks/services/useConnectionHook";
 import { equal } from "utils/objects";
 import ConnectionForm from "views/Connection/ConnectionForm";
+import { ConnectionFormSubmitResult } from "views/Connection/ConnectionForm/ConnectionForm";
 import { FormikConnectionFormValues } from "views/Connection/ConnectionForm/formConfig";
 
+import styles from "./ReplicationView.module.scss";
+
 interface ReplicationViewProps {
   onAfterSaveSchema: () => void;
   connectionId: string;
 }
 
+interface ResetWarningModalProps {
+  onClose: (withReset: boolean) => void;
+  onCancel: () => void;
+  stateType: ConnectionStateType;
+}
+
+const ResetWarningModal: React.FC<ResetWarningModalProps> = ({ onCancel, onClose, stateType }) => {
+  const { formatMessage } = useIntl();
+  const [withReset, setWithReset] = useState(true);
+  const requireFullReset = stateType === ConnectionStateType.legacy;
+  return (
+    <div className={styles.resetWarningModal}>
+      {/* TODO: This should use proper text stylings once we have them available. */}
+      <FormattedMessage id={requireFullReset ? "connection.streamFullResetHint" : "connection.streamResetHint"} />
+      <p>
+        <LabeledSwitch
+          checked={withReset}
+          onChange={(ev) => setWithReset(ev.target.checked)}
+          label={formatMessage({ id: requireFullReset ? "connection.saveWithFullReset" : "connection.saveWithReset" })}
+          checkbox
+        />
+      </p>
+      <div className={styles.resetWarningModalButtons}>
+        <Button onClick={onCancel} secondary>
+          <FormattedMessage id="form.cancel" />
+        </Button>
+        <Button onClick={() => onClose(withReset)}>
```

Comment:
Not sure if it's clear that `onClose` confirms the action. I would it expect that onClose just closes without consequences.